README.md

```diff
@@ -189,6 +189,149 @@ We only did a small refactor, so the build tab running `npm run build` should
 show no errors, and the local page under http://localhost:8080/ should look like
 it did before starting this step.
 
+## Step 3: Modeling data
+
+To build our GitHub activity feed reader, we'll eventually fetch data from this
+endpoint: [https://gh-feed.vercel.app/api](https://gh-feed.vercel.app/api).
+
+This endpoint doesn’t require API keys and works as follows:
+- It accepts two query parameters: `user` (a `string`) and `page` (an `int`).
+  For example: `https://gh-feed.vercel.app/api?user=jchavarri&page=1`
+- It returns an object (let's call it `feed`) with a key `entries`, which is an
+  array of feed entries. Each entry has the following structure:
+  - **content**: The content of the entry, usually HTML as a string.
+  - **links**: An array of associated links, each with:
+    - **href**: The URL the link points to (`string`).
+    - **title**: The link title or description (`string`).
+  - **title**: The entry title (`string`).
+  - **updated**: A timestamp of the last update (`float`).
+
+> [!NOTE]
+> 
+> There are other keys in the returned data, but we will ignore them for
+> simplicity.
+
+In this step, we’ll define some OCaml types to represent this data in our
+application. We’ll use records for the objects and lists for the arrays. Refer
+to the Reason documentation for details:
+- [Records](https://reasonml.github.io/docs/en/record)
+- [Lists](https://reasonml.github.io/docs/en/basic-structures#list)
+
+We need:
+- A type for the feed
+- A type for an entry
+- A type for a link
+
+> [!TIP]
+> 
+> Create a new file `Feed.re` to keep these type definitions separate from the
+> UI code. As we saw before, we are able to use its values from other modules by
+> namespacing it, e.g. `Feed.foo`.
+
+### Step 3 completion check
+
+Ensure your application builds successfully (`Success` in the `npm run watch`
+terminal). We may need to adjust these types later when we decode the data.
+
+## Step 4: Decoding data (the hard way)
+
+Once you have these types defined, we will be adding some code to take the
+values from the JSON and decode them into values of the types we have defined.
+
+First, install the
+[melange-json](https://github.com/melange-community/melange-json) library:
+
+```sh
+opam install melange-json
+```
+
+Then, modify the `libraries` field in the `src/dune` file to include
+`melange-json`:
+
+```dune
+(libraries melange-json reason-react)
+```
+
+Now we’re ready to decode JSON into our types. Add a `Decode` module inside
+`Feed.re` with the following code:
+
+```reason
+module Decode = {
+  let link = json =>
+    Json.Decode.{
+      Link.href: json |> field("href", string),
```

Comment:
So I guess I was supposed to define my types this way?

```reason
module Link = {
  type t = {
    href: string,
    title: string,
  };
};
```
But where did it say that?

Reply:
I ended up removing the submodules and following your approach, also mentioning explicitly the types that need to be created: https://github.com/ahrefs/react-alicante-workshop/pull/3/commits/af667ad639bf0bd3c30e6cd61e4e033744cf94c2.